From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: David Llewellyn-Jones <david@flypig.co.uk>
Date: Sun, 7 Jul 2024 16:28:06 +0100
Subject: [PATCH 44/67] Regenerate moz.build files. JB#53756

Refresh the WebRTC build integration after adding the Sailfish video capture
module.

See the following page for background to these changes:

https://firefox-source-docs.mozilla.org/build/buildsystem/gn.html

The old ESR91 patch carried a large autogenerated moz.build refresh.  ESR115
only needs the remaining Sailfish-specific build plumbing: detect
pkgconfig(geckocamera), use the SFOS video capture sources when gecko-camera is
available, and drop desktop-capture, portal, DRM, GBM, epoxy, and PipeWire GN
directories from EmbedLite builds.

The generated moz.build portions were refreshed with:

$ ./mach build-backend -b GnMozbuildWriter

Co-authored-by: Denis Grigorev <d.grigorev@omprussia.ru>
---
 old-configure.in                              | 10 ++++++++
 .../video_capture_internal_impl_gn/moz.build  | 25 ++++++++++++++-----
 third_party/libwebrtc/moz.build               | 14 +++++++++++
 3 files changed, 43 insertions(+), 6 deletions(-)

diff --git a/old-configure.in b/old-configure.in
index 329c03cfef05..75750ded9d28 100644
--- a/old-configure.in
+++ b/old-configure.in
@@ -1341,6 +1341,16 @@ if test -n "$MOZ_WEBRTC"; then
     if test -n "$MOZ_X11"; then
       MOZ_WEBRTC_X11_LIBS="-lXext -lXdamage -lXfixes -lXcomposite"
     fi
+
+    MOZ_ENABLE_WEBRTC_GECKOCAMERA=
+    PKG_CHECK_MODULES(LIBGECKOCAMERA, geckocamera,
+                     MOZ_ENABLE_WEBRTC_GECKOCAMERA=1,
+                      MOZ_ENABLE_WEBRTC_GECKOCAMERA=)
+    if test "$MOZ_ENABLE_WEBRTC_GECKOCAMERA"; then
+       MOZ_ENABLE_WEBRTC_GECKOCAMERA=1
+       AC_DEFINE(MOZ_ENABLE_WEBRTC_GECKOCAMERA)
+       AC_SUBST(MOZ_ENABLE_WEBRTC_GECKOCAMERA)
+    fi
 fi
 
 AC_SUBST_LIST(MOZ_WEBRTC_X11_LIBS)
diff --git a/third_party/libwebrtc/modules/video_capture/video_capture_internal_impl_gn/moz.build b/third_party/libwebrtc/modules/video_capture/video_capture_internal_impl_gn/moz.build
index cd5557b826da..c4c9e52f290f 100644
--- a/third_party/libwebrtc/modules/video_capture/video_capture_internal_impl_gn/moz.build
+++ b/third_party/libwebrtc/modules/video_capture/video_capture_internal_impl_gn/moz.build
@@ -86,12 +86,25 @@ if CONFIG["OS_TARGET"] == "Linux":
         "rt"
     ]
 
-    UNIFIED_SOURCES += [
-        "/third_party/libwebrtc/modules/video_capture/linux/device_info_linux.cc",
-        "/third_party/libwebrtc/modules/video_capture/linux/device_info_v4l2.cc",
-        "/third_party/libwebrtc/modules/video_capture/linux/video_capture_linux.cc",
-        "/third_party/libwebrtc/modules/video_capture/linux/video_capture_v4l2.cc"
-    ]
+    if CONFIG["MOZ_ENABLE_WEBRTC_GECKOCAMERA"]:
+        LOCAL_INCLUDES += ["/media/libyuv/libyuv/include/"]
+        OS_LIBS += CONFIG["LIBGECKOCAMERA_LIBS"]
+
+        UNIFIED_SOURCES += [
+            "/third_party/libwebrtc/modules/video_capture/sfos/device_info_sfos.cc",
+            "/third_party/libwebrtc/modules/video_capture/sfos/video_capture_sfos.cc"
+        ]
+
+        CXXFLAGS += CONFIG["LIBGECKOCAMERA_CFLAGS"]
+        CXXFLAGS += CONFIG["NSPR_CFLAGS"]
+        CXXFLAGS += CONFIG["MOZ_PIXMAN_CFLAGS"]
+    else:
+        UNIFIED_SOURCES += [
+            "/third_party/libwebrtc/modules/video_capture/linux/device_info_linux.cc",
+            "/third_party/libwebrtc/modules/video_capture/linux/device_info_v4l2.cc",
+            "/third_party/libwebrtc/modules/video_capture/linux/video_capture_linux.cc",
+            "/third_party/libwebrtc/modules/video_capture/linux/video_capture_v4l2.cc"
+        ]
 
 if CONFIG["OS_TARGET"] == "OpenBSD":
 
diff --git a/third_party/libwebrtc/moz.build b/third_party/libwebrtc/moz.build
index 0920c2883cec..4ea76ff67e78 100644
--- a/third_party/libwebrtc/moz.build
+++ b/third_party/libwebrtc/moz.build
@@ -632,6 +632,20 @@ if CONFIG["CPU_ARCH"] == "x86_64" and CONFIG["OS_TARGET"] == "Linux":
         "/third_party/libwebrtc/third_party/pipewire/pipewire_gn"
     ]
 
+if CONFIG["MOZ_EMBEDLITE"]:
+    for desktop_capture_dir in [
+        "/third_party/libwebrtc/modules/desktop_capture/desktop_capture_differ_sse2_gn",
+        "/third_party/libwebrtc/modules/desktop_capture/desktop_capture_gn",
+        "/third_party/libwebrtc/modules/desktop_capture/primitives_gn",
+        "/third_party/libwebrtc/modules/portal/portal_gn",
+        "/third_party/libwebrtc/third_party/drm/drm_gn",
+        "/third_party/libwebrtc/third_party/gbm/gbm_gn",
+        "/third_party/libwebrtc/third_party/libepoxy/libepoxy_gn",
+        "/third_party/libwebrtc/third_party/pipewire/pipewire_gn",
+    ]:
+        if desktop_capture_dir in DIRS:
+            DIRS.remove(desktop_capture_dir)
+
 if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "OpenBSD":
 
     DIRS += [
-- 
2.51.0

